mesos-tools/src/cgcloud/mesos_tools/__init__.py

```diff
@@ -374,3 +375,10 @@ def __prepare_slave_args( self ):
             if attributes:
                 attributes = ';'.join( '%s:%r' % i for i in attributes.items( ) )
                 f.write( "--attributes='%s'" % attributes )
+
+    def __prepare_credentials(self):
+        # Create the credentials file and transfer ownership to mesosbox
+        check_call([sudo, 'touch', '/var/lib/mesos/credentials.txt'])
+        check_call([sudo, 'chown', 'mesosbox:mesosbox', '/var/lib/mesos/credentials.txt'])
+        with open('/var/lib/mesos/credentials.txt', 'w') as f:
+            f.write('toil liot')
```

Comment:
This code runs as root so `open(...,'w')` will have exactly the same effect as running `touch`.

You can also remove the `sudo` from the `chown` invocation.